### ISSUE TITLE
Nested routing

### ISSUE THREAD

**Issue by aleclarson** [alloc/saus#55]
Share multiple layers of parents between two page URLs, where the nearest parent layer could change for other page URLs while the top-level layer is shared by all pages using the same Saus renderer.

**Comment by aleclarson** [comment]
A possible design could be:

```ts
// ./src/node/routes.ts
import { route } from 'saus'

route('/', () => import('./pages/Home'))
  .outlets(() => {
    route('/foo', () => import('./pages/Foo'))
  })
```

Then the `./pages/Home` component would use the `<Outlet>` component (inspired by React Router).

```ts
import { Outlet } from '@saus/react'

export default function Home() {
  return (
	<div>
	  <Outlet />
	</div>
  )
}
```

One remaining question is how to expose the outlet route in a UI-framework-agnostic way.

**Comment by aleclarson** [comment]
HTTP endpoints could also benefit from nesting:
```ts
route('/api')
  .extend(() => {
    route('/posts/:id')
      .get(req => {...})
      .post(req => {...})
  })
```

**Comment by aleclarson** [comment]
Could be cool to have `onRequest` and `onResponse` adhere to the scope of the `extend` callback, too.

```ts
route('/api')
  .extend(() => {
	// These only apply to routes within this calback!
	onRequest(req => {...})
	onResponse((res, req) => {...})

	// Exposed at /api/posts/:id
    route('/posts/:id')
      .get(req => {...})
      .post(req => {...})
  })
```

**Comment by aleclarson** [comment]
The `extend` method now exists in the `next` branch.

But using `onRequest` and `onResponse` within the `extend` callback is not yet.

**Comment by aleclarson** [comment]
Side note: If your `extend` callback is async, you **need** to use the `route` argument passed to it.

```ts
route('/api')
  .extend(async route => {
    const stuff = await loadStuffForGeneratedRoutes()
	for (const route of stuff) {
      route(route.path, route.import)
	}
  })
```

**Comment by aleclarson** [comment]
Closed in favor of #71
